scripts/replace-legacy-export.cjs

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env node
+'use strict'
+
+const { readFileSync, writeFileSync } = require('fs')
+
+// Cleanup the export statement in CJS typings file generated by rollup:
+const legacyTypings = 'build/index.cjs.d.ts'
+const contents = readFileSync(legacyTypings, 'utf8').replace(
+  'export { yargsParser as default };', 'export = yargsParser;'
```

Comment:
Is this a bug in rollup's support for `--exports=default`? If so, is there an open issue this could link to track when the workaround may become unnecessary?

Reply:
@jkrems I wanted to keep the behavior of the CJS module identical to what it is today, with the main export being a callable function, i.e., not `exports.default = yargsParser`. However, for ESM/Deno I wanted to `export default yargsParser`... I wasn't sure how to indicate to rollup that I wanted the `export = yargsParser` behavior, while at the same time using the desired syntax in the source files themselves.

Maybe I should open a ticket on the tsc rollup plugin I'm using?

Reply:
Yeah, it seem like if the `module.exports` object is set to the default export, then `export = defaultExport` would be the correct typings to emit. At least from what I understand about TS type definitions. Not sure how that would affect future ESM consumers (that use vscode code completion) in the presence of named exports etc. but that's not necessarily a problem for today..?